src/core/pratt.ml

```diff
@@ -0,0 +1,64 @@
+(** This module defines functions needed by the Pratt parser of the Pratter
+    library.
+
+    The interface for the Pratter library can be seen at
+    @see <https://forge.tedomum.net/koizel/pratter> *)
+
+open Syntax
+
+module Pratt : sig
+  val parse : Sig_state.t -> Env.t -> p_term -> p_term
+  (** [parse ss env t] Pratt parses term [t], unsugaring infix operators and
+      prefix operators using signature state [ss] and environment [env] to
+      determine which term is an operator, and to build new terms. Note that
+      it doesn't recurse into abstractions or implications and alike. *)
+end = struct
+
+  open Lplib
+  open Pos
+
+  (** [find_sym ss env qid] is [{!val:Sig_state.find_sym} ss qid], but
+      doesn't fail if [qid] is in the environment [env]. It fetches private
+      and protected symbols as well. *)
+  let find_sym : Sig_state.t -> Env.t -> qident -> Terms.sym option =
```

Comment:
Why not unfold this function if it is not used outside this module and if it is called only once?